src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs

```diff
@@ -58,11 +58,12 @@ static CIEnvironmentValues()
                 // Clean branch name
                 if (!string.IsNullOrEmpty(Branch))
                 {
-                    var regex = new Regex(@"^refs\/heads\/(.*)|refs\/(.*)$", RegexOptions.Compiled);
+                    var regex = new Regex(@"^refs\/heads\/(.*)|refs\/tags\/(.*)|refs\/(.*)$", RegexOptions.Compiled);
                     var match = regex.Match(Branch);
-                    if (match.Success && match.Groups.Count == 3)
+                    if (match.Success && match.Groups.Count == 4)
                     {
-                        Branch = !string.IsNullOrWhiteSpace(match.Groups[1].Value) ? match.Groups[1].Value : match.Groups[2].Value;
+                        Branch = !string.IsNullOrWhiteSpace(match.Groups[1].Value) ? match.Groups[1].Value : match.Groups[3].Value;
+                        Tag = match.Groups[2].Value;
```

Comment:
As the number of groups is increasing, you should consider using named groups for better readability